### PR TITLE
feat: [CDS-43587]: Support Human Readable Terraform Plan to be available as JSON Path in CG

### PR DESCRIPTION
**Pull request by akhilpandey1**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38286)

### CODE REVIEW

File: 930-delegate-tasks/src/main/java/io/harness/delegate/expression/TerraformPlanDelegateFunctor.java [hunk 1]
```diff
@@ -38,18 +38,31 @@
...
     }
 
+    @Override
+    public String humanReadableFilePath() {
+      return humanReadableFilePath;
+    }
+
     public void cleanup() {
-      FileUtils.deleteQuietly(new File(jsonPlanFilePath));
+      if (jsonPlanFilePath != null) {
```
Comment (akhilpandey1) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/expression/TerraformPlanDelegateFunctor.java:60]: This was causing NPEs in Execution, so I decided to add a nullcheck

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 9 Issues 
* Code issue 6 Code issues 
* Anti-pattern 3 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/38286)

**Comment by akhilpandey1** [comment]
trigger feature-build

**Comment by akhilpandey1** [comment]
trigger feature-build

**Comment by akhilpandey1** [comment]
trigger feature-build

**Comment by akhilpandey1** [comment]
trigger feature-build

**Comment by akhilpandey1** [comment]
trigger feature-build

**Comment by akhilpandey1** [comment]
trigger checkstyle,trigger pmd,trigger ti0,trigger ti1,trigger ti2,trigger ti3,trigger ti4

**Comment by akhilpandey1** [comment]
trigger feature-build

**Comment by akhilpandey1** [comment]
trigger feature-build

**Comment by akhilpandey1** [comment]
trigger feature-build

**Comment by akhilpandey1** [comment]
trigger feature-build

**Comment by markoharness** [comment]
*[`970-api-services-beans/src/main/java/io/harness/logging/PlanHumanReadableOutputStream.java` line 84 at r6](https://reviewable.io/reviews/harness/harness-core/38286#-NDgMJ6o6QFqa2-ffme_:-NDgMJ6o6QFqa2-ffmea:bdr88n6) ([raw file](https://github.com/harness/harness-core/blob/43d6adc13e99aff1aba3ec2345598ba964b50913/970-api-services-beans/src/main/java/io/harness/logging/PlanHumanReadableOutputStream.java#L84)):*
> ```Java
>       try (FileInputStream fis = new FileInputStream(getTfHumanReadablePlanLocalPath());
>            GZIPInputStream tfDecompressedInputStream = new GZIPInputStream(fis);) {
>         byte[] buffer = new byte[10];
> ```

I would increase the buffer size to at least 4096. It will give much better performance.

**Review by markoharness — COMMENTED** [review]
> Reviewed 7 of 13 files at r1, 1 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38286)* status: 5 file(s) to review. 0 of 1 approvals obtained.

**Comment by akhilpandey1** [comment]
trigger feature-build

**Review by markoharness — COMMENTED** [review]
> Reviewed 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38286)* status: 0 of 1 approvals obtained.

**Comment by akhilpandey1** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/expression/TerraformPlanDelegateFunctor.java` line 60 at r1](https://reviewable.io/reviews/harness/harness-core/38286#-NDTKMGUzZp_SxcIApTK-r1-60:-NDm79PQ8drjbG5aJHYi:bb6ywmx) ([raw file](https://github.com/harness/harness-core/blob/305b5b97674ff0eae8e74e856792c5a3e7c36523/930-delegate-tasks/src/main/java/io/harness/delegate/expression/TerraformPlanDelegateFunctor.java#L60)):*
<details><summary><i>Previously, akhilpandey1 (Akhilesh Pandey) wrote…</i></summary><blockquote>

This was causing NPEs in Execution, so I decided to add a nullcheck
</blockquote></details>

LGTM

**Review by akhilpandey1 — COMMENTED** [review]
> Reviewed 10 of 13 files at r1, 2 of 3 files at r2, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38286)* status: 1. unresolved discussions. 0 of 1 approvals obtained.

**Comment by akhilpandey1** [comment]
Is it new task type? (If yes, no need to answer 2 then)
- No

Did you add a new class which is added in kryo serialization
-No

Did you make changes to the code in a backward-compatible manner?
-Yes

will the new version of the delegate receive old version task and process?
-Yes

If new version of the manager receives response for the old task will it be able to handle it successfully?
-Yes

Ng backward compatibility → did you make a change in delegate task which will be created in ng services via grpc call to manager
-Yes

cc: @yogesh-chauhan

**Review by yogesh-chauhan — APPROVED** [review]

**Comment by yogesh-chauhan** [comment]
trigger codebasehash

**Pull request merged**